gradio/image_utils.py

```diff
@@ -39,10 +46,21 @@ def format_image(
 
 
 def save_image(y: np.ndarray | _Image.Image | str | Path, cache_dir: str):
+    # numpy gets saved to png as default format
+    # PIL gets saved to its original format if possible
     if isinstance(y, np.ndarray):
         path = processing_utils.save_img_array_to_cache(y, cache_dir=cache_dir)
     elif isinstance(y, _Image.Image):
-        path = processing_utils.save_pil_to_cache(y, cache_dir=cache_dir)
+        fmt = y.format
+        try:
+            path = processing_utils.save_pil_to_cache(
+                y, cache_dir=cache_dir, format=fmt  # type: ignore
+            )
+        # Catch error if format is not supported by PIL
+        except Exception:
```

Comment:
looks like its a KeyError that PIL raises for invalid formats
```suggestion
        except KeyError:
```